ocaml/alerts/certificate/certificate_check.ml

```diff
@@ -42,24 +42,36 @@ let generate_alert epoch (host, expiry) =
 
 let execute rpc session previous_messages (host, alert) =
   let host_uuid = XenAPI.Host.get_uuid rpc session host in
-  let obsolete_messages =
-    previous_messages
-    |> List.filter_map (fun (ref, record) ->
-           if record.API.message_obj_uuid = host_uuid then
-             Some ref
-           else
-             None)
+  let host_messages =
+    List.filter
+      (fun (_, record) -> record.API.message_obj_uuid = host_uuid)
+      previous_messages
   in
-  List.iter
-    (fun self -> XenAPI.Message.destroy rpc session self)
-    obsolete_messages ;
-  match alert with
-  | None ->
-      ()
-  | Some (message, (alert, priority)) ->
+  let is_current (ref, record) =
+    match alert with
+    | None ->
+        false
+    | Some (body, (name, priority)) ->
+        record.API.message_body = body
+        && record.API.message_name = name
+        && record.API.message_priority = priority
+  in
+  let _, outdated_messages = List.partition is_current host_messages in
```

Comment:
This could be changed by a `List.filter` if it hurts readability. These alerts will most probably be affected by adding the metadata of the whole chain of certificates in the database.

Reply:
I am not seeing the advantage of using partition

Reply:
I also don't see the advantage, since you drop the first return value.